### PR TITLE
Finish "Create aligned data" button in manual axis alignment UI

### PR DESCRIPTION
**Pull request by yijiang1**
Finishing up the "Create aligned data" button in manual axis alignment UI.
As required by @Hovden , The 'create align button' now "create a shift and rotate python data transform in the pipeline. The shift will be a global shift (applied first) and then a global rotation (applied second)."

@mathturtle  I'm not sure what's the optimal way to assess the python scripts, but my code works.

Also, the existing "shift uniformly" python function is not suitable for alignment, because it assumes periodic boundary condition and only takes integer inputs. I'll write a new one later.

### CODE REVIEW

**Comment by mathturtle** [comment]
I don't really think we want to call python from a C++ operation like this.  Ideally we should write C++ code for the rotate and shift that you want to apply.  I think vtkImageReslice is able to do what you want, but it is complicated to use...

**Comment by Hovden** [comment]
Right now placing it in the python pipeline is good. It provides a transparent workflow. We can revisit this as a c++ interface later. (similarly, we need to address the manual translation alignment c++ approach since cloning is not the best approach).

**Comment by mathturtle** [comment]
@yijiang1 Let me add some helper function in that you can use.  We don't really want to duplicate all the logic for python operators here...

**Comment by yijiang1** [comment]
@mathturtle Sure. Thanks

**Comment by yijiang1** [comment]
New changes:
1. Use the new helper functions introduced in #261 to read python scripts.
2. Added a new python function "Shift3D" to shift aligned data. It does not warp the image and can do sub-pixel shifts.
3. Rotation angle spinbox's step size is set to 0.5.

File: tomviz/RotateAlignWidget.cxx [hunk 1]
```diff
@@ -509,9 +510,29 @@ void RotateAlignWidget::onFinalReconButtonPressed()
...
+  //Apply shift (in y-direction) first
+  QMap<QString, QString> substitutions;
+  substitutions.insert("###SHIFT###",
+                       QString("SHIFT = [%1, %2, %3]").arg(0)
+                       .arg(-this->Internals->Ui.rotationAxis->value()).arg(0));
+
+  QString scriptLabel = "Shift";
+  QString scriptSource = readInPythonScript("Shift3D");
+  AddPythonTransformReaction::addPythonOperator(this->Internals->Source, scriptLabel, scriptSource, substitutions);
+
```
Comment (mathturtle) [inline tomviz/RotateAlignWidget.cxx]: You may want to add
```c++
substitutions.clear();
```
call here.  It doesn't really matter unless `###SHIFT###` appears in your second script though.

File: tomviz/RotateAlignWidget.h [hunk 2]
```diff
@@ -49,6 +49,7 @@ protected slots:
 
 private:
   Q_DISABLE_COPY(RotateAlignWidget)
+  QString readScript(const QString &scriptName);
```
Comment (mathturtle) [inline tomviz/RotateAlignWidget.h]: You can take this out now that the helper functions are in.

File: tomviz/RotateAlignWidget.cxx (hunk 1 above)
Comment (yijiang1) [inline tomviz/RotateAlignWidget.cxx]: Yeah it doesn't matter. But it's good to know the clear function.

File: tomviz/RotateAlignWidget.h (hunk 2 above)
Comment (yijiang1) [inline tomviz/RotateAlignWidget.h]: Good catch!

**Pull request merged**